src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/jpa/AuditRepository.kt

```diff
@@ -17,7 +17,17 @@ interface AuditRepository : PagingAndSortingRepository<AuditEvent, UUID> {
     select 
         ae from AuditEvent ae 
     where
-        ((cast(:startDate as string) is null or cast(:endDate as string) is null) or (ae.when between :startDate and :endDate) )
+        ((cast(:startDate as string) is null and cast(:endDate as string) is null)  
+          or (cast(:startDate as string) is not null 
+              and cast(:endDate as string) is not null
+              and (ae.when between :startDate and :endDate)) 
+          or (cast(:startDate as string) is not null 
+              and cast(:endDate as string) is null
+              and ae.when >=:startDate ) 
+          or (cast(:startDate as string) is null 
+              and cast(:endDate as string) is not null
+              and ae.when <=:endDate )  
+        )
```

Comment:
whilst this works, you can simplify this further by just doing :
(cast(:startDate as string) is null or ae.when >=:startDate ) 
and (cast(:endDate as string) is null or ae.when <=:endDate )

Reply:
Hmm, I get you Karen! Choice of AND vs OR. Mine is more defensive, descriptive, maps to the 3 date param test scenarios and not less performant either!!! But, fair shout! Thumbs up!